lib/participant_profile_deduplicator.rb

```diff
@@ -101,6 +102,12 @@ def reconcile_schedules!
     update_primary_profile_schedule(profile_with_correct_schedule.latest_induction_record.schedule)
   end
 
+  def update_mentored_ects!
+    return unless [primary_profile, duplicate_profile].all?(&:mentor?)
+
+    duplicate_profile.mentee_profiles.update_all(mentor_profile_id: primary_profile.id)
```

Comment:
I think we'll also need to update the induction records as well with the new profile? see `Induction::ChangeMentor.call(induction_record:, mentor_profile: new_mentor_profile)` as might help with this

Reply:
Agreed, we need to update any induction record whose `mentor_profile_id` matches the `duplicate_profile.id`.
Actually I would only update the induction records and not the `duplicate_profile.mentee_profiles...` because that is a legacy association not used/updated anymore.

Reply:
```suggestion
    InductionRecord.where(mentor_profile_id: duplicate_profile.id).update_all(mentor_profile_id: primary_profile.id)
```

Reply:
I think it's safer to update both to avoid any invalid linking. I think @ethax-ross has the IR update below :)

Reply:
It's fine. I think what we should do is to delete those legacy mentee associations once and for all.
And yes, sorry I didn't see the update of the induction records below!.